pymess/backend/dialer/daktela.py

```diff
@@ -44,13 +44,18 @@ def _update_dialer_states(self, messages):
             })
             resp_message_state = resp_json['result']['statuses'][0]['name'] if len(
                 resp_json['result']['statuses']) else resp_json['result']['action']
-            whole_message_heard = resp_json['result']['customFields']['whole_message_heard'][0] == 'Yes'
+            custom_fields = resp_json['result']['customFields']
+            tts_processed = custom_fields['ttsprocessed'][0]
+            whole_message_heard = custom_fields['whole_message_heard'][
+                                      0] == 'Yes' if 'whole_message_heard' in custom_fields and len(
+                custom_fields['whole_message_heard']) else False
```

Comment:
whole_message_heard = 'whole_message_heard' in custom_fields and custom_fields['whole_message_heard'] and  custom_fields['whole_message_heard'][0] == 'Yes'

Reply:
OK